EMS/core-bundle/src/Core/ContentType/FieldType/FieldTypeTreeItem.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CoreBundle\Core\ContentType\FieldType;
+
+use EMS\CoreBundle\Entity\FieldType;
+
+/**
+ * @implements \IteratorAggregate<FieldTypeTreeItem>
+ */
+class FieldTypeTreeItem implements \IteratorAggregate
+{
+    private FieldTypeTreeItemCollection $children;
+    private string $name;
+    private ?FieldTypeTreeItem $parent = null;
+
+    public function __construct(private readonly FieldType $fieldType)
+    {
+        $this->name = $this->fieldType->getName();
+        $this->children = new FieldTypeTreeItemCollection();
```

Comment:
Why not a recursif __construct?

Reply:
You mean like this: https://github.com/ems-project/elasticms/pull/788/commits/eba66f0adc4b1687ecfa837532dcfd0feed934c4